stdlib/Profile/src/Profile.jl

```diff
@@ -22,6 +22,7 @@ appended to an internal buffer of backtraces.
 """
 macro profile(ex)
     return quote
+        check_init() # if the profile buffer hasn't been initialized, initialize with default size
         try
             status = start_timer()
             if status < 0
```

Comment:
I would suggest we move all of these lines into `start_timer`, as there isn't really a reason to outline them explicitly into the user's macro